moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,268 @@
+        self.app_name = self.container_name.split('/')[-1]
+        self.app_ip = None
+        common.print_debug(self.icontainer, self.app_name)
+
+    def get_container_url(self):
+        """
+        It returns actual URL link string to container, It is same as URL
+
+        :return: str
+        """
+        return self.icontainer
+
+    def get_docker_instance_name(self):
+        """
+        Return docker instance name what will be used inside docker as docker image name
+        :return: str
+        """
+        return self.container_name
+
+    def _app_exists(self):
+        """
+        It checks if an application already exists in OpenShift environment
+        :return: True, application exists
+                 False, application does not exist
+        """
+        oc_status = self.runHost("oc status", ignore_status=True)
+        if 'dc/%s' % self.app_name in oc_status.stdout:
+            common.print_info("Application already exists.")
+            return True
+        oc_services = self.runHost("oc get services -o json")
+        json_svc = self._convert_string_to_json(oc_services.stdout)
+        if not json_svc["items"]:
+            return False
+        return True
+
+    def _convert_string_to_json(self, string):
+        """
+        It converts a string to json format
+        :param string: String to format to json
+        :return: json output
+        """
+        return json.loads(string)
+
+    def _remove_apps_from_openshift_namespaces(self, oc_service="svc"):
+        """
+        It removes an application from specific "namespace" like svc, dc, is.
+        :param oc_service: Service from which we would like to remove application
+        """
+        # Check status of svc/dc/is
+        oc_get = self.runHost("oc get %s" % oc_service, ignore_status=True).stdout
+        # The output is like
+        # dovecot     172.30.1.1:5000/myproject/dovecot     latest    15 minutes ago
+        # memcached   172.30.1.1:5000/myproject/memcached   latest    13 minutes ago
+
+        app_found = [x for x in oc_get.split('\n') if x.startswith(self.app_name)]
+        if app_found:
+            # If application exists in svc / dc / is namespace, then remove it
+            oc_delete = self.runHost("oc delete %s/%s" % (oc_service, self.app_name), ignore_status=True)
+
+    def _app_remove(self):
+        """
+        Function removes an application from all OpenShift namespaces like 'svc', 'dc', 'is'
+        """
+        if self._app_exists():
+            # TODO get info from oc status and delete relevat svc/dc/is
+            for ns in ['svc', 'dc', 'is']:
+                self._remove_apps_from_openshift_namespaces(ns)
+
+    def _create_app(self):
+        """
+        It creates an application in OpenShift environment
+        """
+        # Switching to system user
+        oc_new_app = self.runHost("oc new-app %s --name=%s" % (self.container_name,
```

Comment:
I think you should add a label here to mark the deployment being owned by mtf: `oc new-app -l mtf_testing=true` or something like that; and then operate on objects which have this label.

Reply:
@phracek @TomasTomecek yep, similar to `docker container ...` commands and then you will just use this tag. otherwise I expect that this __iddentifier__ will be stored in OpenShiftHelper class like `self.jmeno` for `ContainerHelper`